arch/arc/core/irq_manage.c

```diff
@@ -81,41 +81,111 @@ void z_arc_firq_stack_set(void)
 }
 #endif
 
+/*
+ * ARC CPU interrupt controllers hierarchy.
+ *
+ * Single-core (UP) case:
+ *
+ *   --------------------------
+ *   |  CPU core 0            |
+ *   --------------------------
+ *   |  core 0 (private)      |
+ *   |  interrupt controller  |
+ *   --------------------------
+ *               |
+ *       [interrupt lines]
+ *
+ *
+ * SMP case:
+ *
+ *   --------------------------         --------------------------
+ *   |  CPU core 0            |         |  CPU core 1            |
+ *   --------------------------         --------------------------
+ *   |  core 0 (private)      |         |  core 1 (private)      |
+ *   |  interrupt controller  |         |  interrupt controller  |
+ *   --------------------------         --------------------------
+ *         |      |                             |      |
+ *         |    [core 0 private interrupts]     |    [core 1 private interrupts]
+ *         |                                    |
+ *        ---------------------------------------------
+ *        |     IDU (Interrupt Distribution Unit)     |
+ *        ---------------------------------------------
+ *                            |
+ *               [common (shared) interrupts]
+ */
```

Comment:
Is it already documented somewhere how private and shared interrupts should be wired up (common vectors start at ARC_CONNECT_IDU_IRQ_START)?
This is not a hardware integration restriction, but a software imposed limitation, right? So it could be that hardware integrators to choose a different way of wiring up private/common interrupts.

Reply:
Discussed internally - for now no changes required.